addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoRoomHandler.java

```diff
@@ -0,0 +1,289 @@
+            if (brainApi == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                return;
+            }
+
+            NeeoUtil.checkInterrupt();
+            final NeeoRoomProtocol protocol = new NeeoRoomProtocol(
+                    new StatefulHandlerCallback(new NeeoHandlerCallback() {
+
+                        @Override
+                        public void statusChanged(ThingStatus status, ThingStatusDetail detail, String msg) {
+                            updateStatus(status, detail, msg);
+                        }
+
+                        @Override
+                        public void stateChanged(String channelId, State state) {
+                            updateState(channelId, state);
+                        }
+
+                        @Override
+                        public void setProperty(@NonNull String propertyName, String propertyValue) {
+                            getThing().setProperty(propertyName, propertyValue);
+                        }
+
+                        @Override
+                        public void scheduleTask(Runnable task, long milliSeconds) {
+                            scheduler.schedule(task, milliSeconds, TimeUnit.MILLISECONDS);
+                        }
+                    }), brainApi, config.getKey());
+            NeeoUtil.close(roomProtocol.getAndSet(protocol));
+
+            NeeoUtil.checkInterrupt();
+            updateStatus(ThingStatus.ONLINE);
+
+            NeeoUtil.checkInterrupt();
+
+            NeeoUtil.cancel(refreshTask.getAndSet(scheduler.scheduleAtFixedRate(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        refreshState();
+                    } catch (Exception e) {
+                        logger.debug(">>> hey!", e);
+                    }
+                }
+            }, 0, 1, TimeUnit.SECONDS)));
+        } catch (IOException e) {
+            logger.debug("IOException during initialization", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Room " + config.getKey() + " couldn't be found");
+        } catch (InterruptedException e) {
+            logger.debug("Initialization was interrupted", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Initialization was interrupted");
+        }
+    }
+
+    /**
+     * Refreshes the state of the room by calling {@link NeeoRoomProtocol#refreshState()}
+     */
+    private void refreshState() {
+        try {
+            NeeoUtil.checkInterrupt();
+            final NeeoRoomProtocol protocol = roomProtocol.get();
+            if (protocol != null) {
+                NeeoUtil.checkInterrupt();
+                protocol.refreshState();
+            }
+        } catch (InterruptedException e) {
+            logger.debug("Refresh State was interrupted");
```

Comment:
In most use cases you're not allowed to suppress an `InterruptedException`. If you want to log it, you should or re-throw set interrupt status: `Thread.currentThread().interrupt();`